File: src/main/java/com/github/mystery2099/voxlib/combination/VoxelShapeModifier.java
package com.github.mystery2099.voxlib.combination;

import net.minecraft.world.phys.shapes.VoxelShape;
import static com.github.mystery2099.voxlib.combination.VoxelAssembly.and;

public class VoxelShapeModifier {
    private VoxelShape storedShape;

    VoxelShapeModifier(VoxelShape stored) {
        storedShape = stored;
    }

    public VoxelShape append(VoxelShape shape) {
        storedShape = and(storedShape, shape);
        return storedShape;
    }

    public VoxelShape getStoredShape() { return storedShape; }
    public void setStoredShape(VoxelShape s) { this.storedShape = s; }
}

File: src/main/kotlin/com/github/mystery2099/voxelshapeutils/combination/VoxelShapeModifier.kt
package com.github.mystery2099.voxelshapeutils.combination

import com.github.mystery2099.voxelshapeutils.combination.VoxelShapeCombining.plus
import net.minecraft.util.shape.VoxelShape

class VoxelShapeModifier internal constructor(private var baseShape: VoxelShape) {
    /**
     * Case
     *
     * @param condition
     * @return
     */
    infix fun VoxelShape.case(condition: Boolean): VoxelShape {
        return append(this, condition)
    }

    /**
     * Append
     *
     * @param shape
     * @param condition
     * @return
     */
    fun append(shape: VoxelShape, condition: Boolean = true): VoxelShape {
        if (condition) baseShape += shape
        return baseShape
    }
}